### ISSUE TITLE
Change Request, ItineraryRequest and Itinerary models date storage to DateTimeOffset...

### ISSUE THREAD

**Issue by mgmccarthy** [HTBox/allReady#1386]
@tonysurma, you can assign this to me for now, but we could potentially have multiple people work on this.

after a conversation yesterday afternoon with @dpaquette, it looks like we need to standardize on using DateTimeOffset throughout the system.

Right now, Request, ItineraryRequest and Itinerary all use DateTime properties, and the way the properties are named, it's not clear whether there are intended to hold UTC values or not... 

I have not traced through the code to see how these date time values are populated from the UI..

This is a fairly large refactoring that will include migrations, changing ViewModels, razor views, and having to use the `IConvertDateOffset` provider to convert the incoming DateTimeOffset on a write operation to the correct DateTimeOffset based on a TimeZoneId.

Thus, it's probably better to do this refactoring one model and a time, starting with Itinerary.

On a side note, the soon to be pushed PR for part of Issue #1032 and #1084 are incorrectly trying to do date and time based calcs on DateTime values instead of DateTimeOffset values... once this refactoring is done, these calculations (which sends sms texts to people, so we HAVE to get it right) can correctly try to localize on a Request's Intinerary's Campaign's TimeZone to control the sending of SMS messages.

**Comment by dpaquette** [comment]
@mgmccarthy let me know when you want to chat about this one now that the timezone / offset binding stuff has been merged

**Comment by mgmccarthy** [comment]
@dpaquette, I'll reach out to you on Slack/Skype and we can discuss. We might not have to refactor all three of these.

**Comment by mgmccarthy** [comment]
@tonysurma, after a convo with @dpaquette  tonight, this issue can be closed because it no longer applies.

I will be working on an update to the Hangfire jobs in conjunction with the Itinerary's Event's TimeZoneId, but I'll create a separate issue for that.

Thanks!

**Comment by mgmccarthy** [comment]
ping @tonysurma, this can be closed as it's no longer relevant.
